src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,12 +36,37 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * The extension settings include a name, host address, and port.
+     */
+    private ExtensionSettings settings;
+
+    /**
+     * Empty constructor to fulfill abastract class requirement
      */
     protected BaseExtension() {
```

Comment:
I don't think we need this constructor any more now that we have others, and we want to require subclasses to provide settings.